### PR TITLE
[network-gateway] migrate network-gateway module to distroless

### PR DESCRIPTION
**Pull request by artbin**
## Description

`network-gateway` module is based on a distroless image.

## Why do we need it, and what problem does it solve?

We'd like to base our images on a distroless image.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: network-gateway
type: feature
summary: network-gateway module is based on a distroless image.
impact_level: default
```

### CODE REVIEW

**Comment by artbin** [comment]
Manual testing
==============

Enable `network-gateway` module:
```bash
kubectl create -f - <<EOF
apiVersion: deckhouse.io/v1alpha1
kind: ModuleConfig
metadata:
  name: network-gateway
spec:
  settings:
    nodeSelector:
      network-gateway: "true"
    subnet: 10.100.0.6/24
    publicAddress: 9.9.9.9
  enabled: true
  version: 1
EOF
```

Get dhcp pod logs:
```bash
kubectl -n d8-network-gateway logs dhcp-0
```

Output:
```
dnsmasq: started, version 2.89 cachesize 150
dnsmasq: compile time options: IPv6 GNU-getopt no-DBus no-UBus no-i18n IDN2 DHCP DHCPv6 no-Lua TFTP no-conntrack ipset no-nftset auth cryptohash DNSSEC loop-detect inotify dumpfile
dnsmasq: reading /etc/resolv.conf
dnsmasq: using nameserver 10.222.0.10#53
dnsmasq: read /etc/hosts - 8 names
```

Get snat pod iptables rules:
```bash
kubectl exec -i -t -n d8-network-gateway snat-64w2f -- iptables -w -t nat -n --list D8-NETWORK-GATEWAY-SNAT
```

Output:
```
Chain D8-NETWORK-GATEWAY-SNAT (1 references)
target     prot opt source               destination         
RETURN     all  --  0.0.0.0/0            10.222.0.0/16        /* d8_netgw_cfgHash_9402a66bcae6868bd2debe8082af27b1855f31880b0a537be65b55a65bb375a9 */
RETURN     all  --  0.0.0.0/0            10.111.0.0/16        /* d8_netgw_cfgHash_9402a66bcae6868bd2debe8082af27b1855f31880b0a537be65b55a65bb375a9 */
RETURN     all  --  0.0.0.0/0            10.100.0.0/24        /* d8_netgw_cfgHash_9402a66bcae6868bd2debe8082af27b1855f31880b0a537be65b55a65bb375a9 */
RETURN     all  --  0.0.0.0/0            224.0.0.0/4          /* d8_netgw_cfgHash_9402a66bcae6868bd2debe8082af27b1855f31880b0a537be65b55a65bb375a9 */
SNAT       all  --  0.0.0.0/0            0.0.0.0/0            /* d8_netgw_cfgHash_9402a66bcae6868bd2debe8082af27b1855f31880b0a537be65b55a65bb375a9 */ to:9.9.9.9
```

File: ee/modules/450-network-gateway/images/dnsmasq/werf.inc.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    before: install
+  - artifact: {{ $.ModuleName }}/{{ $.ImageName }}-binaries-artifact
+    add: /usr/lib64/python3
+    to: /usr/lib64/python3
+    before: install
+  - artifact: {{ $.ModuleName }}/{{ $.ImageName }}-binaries-artifact
+    add: /usr/lib64/python3.9
+    to: /usr/lib64/python3.9
+    before: install
+shell:
```
Comment (RomanenkoDenys) [inline ee/modules/450-network-gateway/images/dnsmasq/werf.inc.yaml:29]: move it to artifact

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some nits

File: ee/modules/450-network-gateway/images/dnsmasq/werf.inc.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    before: install
+  - artifact: {{ $.ModuleName }}/{{ $.ImageName }}-binaries-artifact
+    add: /usr/lib64/python3
+    before: install
+  - artifact: {{ $.ModuleName }}/{{ $.ImageName }}-binaries-artifact
+    add: /usr/lib64/python3.9
+    before: install
+  - artifact: {{ $.ModuleName }}/{{ $.ImageName }}-binaries-artifact
+    add: /usr/local/lib/python3/site-packages
+    before: install
```
Comment (RomanenkoDenys) [inline ee/modules/450-network-gateway/images/dnsmasq/werf.inc.yaml:30]: Please add this dirs to relocated.

File: ee/modules/450-network-gateway/images/snat/werf.inc.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    to: /usr/lib64/python3.9
+    before: setup
+  - artifact: {{ $.ModuleName }}/{{ $.ImageName }}-binaries-artifact
+    add: /lib64/iptables
+    to: /lib64/iptables
+    before: setup
+  - image: common/iptables-wrapper
+    add: /iptables-wrapper
+    to: /iptables-wrapper
+    before: setup
```
Comment (RomanenkoDenys) [inline ee/modules/450-network-gateway/images/snat/werf.inc.yaml:42]: same

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some nits

**Review by RomanenkoDenys — APPROVED** [review]

**Pull request merged**